### ISSUE TITLE
[Bug] added cards counting

### ISSUE THREAD

**Issue by grubin30** [ankidroid/Anki-Android#8443]
###### Reproduction Steps

1. Run AnkiDroid for a while, learning 6 new cards per day and reviewing many more.
2. Open the Statistics page showing the default view of one month, view Added and the Forecast

###### Expected Result

Accurate counting and reasonably accurate forecasting

###### Actual Result

The Added section reports less than 1 new card per day learned (23 for the month), even though I've been learning 6 new cards per day for months (about 180 for the month, and often more before that). I check this daily, and it is consistent throughout the month.

The Forecast shows continually declining numbers of cards over the month (from about 80 to about 55), even though my rate of cards is usually steady in the range of 85-100+. The Forecast seems pretty reasonable if I wasn't learning new cards, which suggests that it has the same problem. 

My accuracy on all cards is about 90% on this deck.

###### Debug info
Refer to the [support page](https://ankidroid.org/docs/help.html) if you are unsure where to get the "debug info".
AnkiDroid Version = 2.14.6

Android Version = 10

ACRA UUID = 1426d222-3dca-40d3-924a-0428b146db80

Scheduler = std

Crash Reports Enabled = true

###### Research
*Enter an [x] character to confirm the points below:*
- [x ] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- - [x ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x ] I have searched for similar existing issues here and on the user forum
- - [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))
###

**Comment by tapish2000** [comment]
I would like to see what's going wrong

**Comment by david-allison-1** [comment]
> I would like to see what's going wrong

Feel free to diagnose and post the results here

**Comment by tapish2000** [comment]
@grubin30 `Added` Section shows only the cards which are newly added, there are two fields in that section - `Total` and `Average`. When you write:
> The Added section reports less than 1 new card per day learned (23 for the month), even though I've been learning 6 new cards per day for months (about 180 for the month, and often more before that).

You say, Added section reports less than 1 new card per day "learned", I think it shows "added", and by less than 1 new card, are you talking about the `Average` field of `Added` Section?

**Comment by grubin30** [comment]
Sorry, I entered the report on my phone, so details were hard to get down exactly. The Added section shows 23 Total, which is under the header "1 Month", so I presume that's 23 for the month. The Average is 0.7 cards/day. As I mentioned, I'm adding 6 new cards per day.

**Comment by tapish2000** [comment]
Do you have some associated screenshots? Because for this to verify, I need to add new cards daily consistently to get the desired output.

**Comment by grubin30** [comment]
![ankid-stats](https://user-images.githubusercontent.com/81839817/113711809-67387e00-96b3-11eb-8170-c810d1664917.png)
![ankid-forecast](https://user-images.githubusercontent.com/81839817/113711813-6869ab00-96b3-11eb-9cb0-c5f90fa82d99.png)
![ankid-actual](https://user-images.githubusercontent.com/81839817/113711821-6a336e80-96b3-11eb-819d-ab8d3dd8f2c4.png)
Here are three screenshots. 
1. shows the statistics screen from my session yesterday: 1 Month Total is 22 cards Added. I added 6 cards, which I gather is what becomes the 15 Learn cards. 
2. shows the forecast screen. Even the highest bar, for the next day's forecast at 85, is about 10 lower than the number of Review cards I actually had this morning. The Young cards drop off a lot during the month, suggesting that the number of new cards being added is not accurately estimated at all.
3. shows the review graph for the past 30 days. I study over 75 review cards nearly every day, way above the forecasts. I have lots of Young reviews each day, plus the blue bars show my Added cards, which are consistent because I add 6 new cards each day.
I hope this helps. Thanks!

**Comment by tapish2000** [comment]
Okay, thanks for the screenshots. I will have a look at all of these, one by one.

**Comment by mikehardy** [comment]
there are some settings related to statistics forecasting, it would be interesting to see how accurate (or inaccurate) the forecasts are with different settings

in settings -> advanced -> advanced statistics I have these things set:
- enable advanced statistics
- compute first n days: 8
- precision of computation: 95%
- number of iterations: 2

my forecasts seem fairly accurate ?

If the non-advanced forecasts are inaccurate and the only reason we did not do advanced statistics all the time was because it was computationally expensive, we might be able to revisit that since it is not 2008 anymore and phones are quite a bit more powerful...

**Comment by tapish2000** [comment]
Yeah, I enabled these settings and checked the forecast, seems fine to me too.

@mikehardy Should I enable this setting by default then? If some user doesn't want to use this, it can be disabled by them manually.

**Comment by mikehardy** [comment]
Maybe? I don't know why they aren't enabled yet. Check git blame maybe? Is it a performance issue? What settings are good? Etc

**Comment by david-allison-1** [comment]
Advanced settings are a deviation from Anki Desktop's (legacy) settings, so they're disabled by default

**Comment by tapish2000** [comment]
If they have to be disabled here, then what we can do here is, ask the users to enable the `advanced statistics` option whenever they visit the Statistics page the first time. If the users want they can enable it, otherwise they can leave it as it is, but at least they will know that there is some setting possible.

**Comment by david-allison-1** [comment]
I don't think enabling legacy statistics will be the answer.

----

In my mind, this report is either:

* We haven't explained well enough what the concept means, and need to ask exploratory questions to find out:
   * What is a month? Do we define it in the same way as the user? When does a day start? Is DST an issue?
   * What are the reviewer settings? Is "bury related cards" set? Do they have a review limit
   * etc...
* There's a bug, and we differ from Anki Desktop's legacy statistics (hold shift and press stats).

----

I think the first stage in this would be to obtain the `collection.anki2`, and compare the statistics to what Anki Desktop is showing.

If we match Anki Desktop, then we can explain the concepts

If we don't match Anki Desktop, then we have data and can write regression tests and fix the bug

**Comment by grubin30** [comment]
Just an update on this, sorry for the delay. Some new screenshots from May and now: [https://photos.app.goo.gl/EFNEztF3fDHrdwzEA](https://photos.app.goo.gl/EFNEztF3fDHrdwzEA)  

I went in after these posts and turned on the advanced statistics. That certainly fixed the issue for the forecast graph, which now goes up instead of down. It doesn't seem to burden my phone to compute all 30 days, but I changed it to 10, since it didn't noticeably change the results. 

I still notice that the number shown under "ADDED" on the first statistics page is very low, e.g. 0.8 cards/day when I had a habit of adding 6 cards/day. On another deck, it shows 0 cards added, even though I've added 5 cards even today and probably about 25 cards this month. Also, the number of reviews in the FORECAST is also very low, at 1285, despite the graph showing semi-accurately that I will instead have at least 100+/day (actually going up). So I think that the statistics on that page are still not accurate.

Minor note: I happen to have a lot of older cards in my larger deck where the "GOOD" button would make the interval 1.5-2.0x the current interval, e.g. from 1.1 years to 2.2 years. So I've been hitting the "HARD" button at a relatively high frequency, but only for these older cards, but this raises the forecasted reviewing rate for the upcoming 30 days. I get that fixing this would be hard, but I want to explain why my graph is going up even though I stopped adding new cards to this deck a few days ago (and it's particularly bad in May).
